migrations/kdvh/import.go

```diff
@@ -0,0 +1,324 @@
+			// return count, nil
+		}
+	}
+
+	if !(config.Skip == "flags") {
+		if err := lard.InsertFlags(ts, FLAGS_TABLE, FLAGS_COLS, pool, tsInfo.logstr); err != nil {
+			slog.Error(tsInfo.logstr + "failed flag bulk insertion - " + err.Error())
+		}
+	}
+
+	return count, nil
+}
+
+func getStationNumber(station os.DirEntry, stationList []string) (int32, error) {
+	if !station.IsDir() {
+		return 0, errors.New(fmt.Sprintf("%s is not a directory, skipping", station.Name()))
+	}
+
+	if stationList != nil && !slices.Contains(stationList, station.Name()) {
+		return 0, errors.New(fmt.Sprintf("Station %v not in the list, skipping", station.Name()))
+	}
+
+	stnr, err := strconv.ParseInt(station.Name(), 10, 32)
+	if err != nil {
+		return 0, errors.New("Error parsing station number:" + err.Error())
+	}
+
+	return int32(stnr), nil
+}
+
+func getElementCode(element os.DirEntry, elementList []string) (string, error) {
+	elemCode := strings.ToUpper(strings.TrimSuffix(element.Name(), ".csv"))
+
+	if elementList != nil && !slices.Contains(elementList, elemCode) {
+		return "", errors.New(fmt.Sprintf("Element '%s' not in the list, skipping", elemCode))
+	}
+
+	if elemcodeIsInvalid(elemCode) {
+		return "", errors.New(fmt.Sprintf("Element '%s' not set for import, skipping", elemCode))
+	}
+	return elemCode, nil
+}
+
+func getTimeseriesID(tsInfo *TimeseriesInfo, pool *pgxpool.Pool) (int32, error) {
+	label := lard.Label{
+		StationID: tsInfo.station,
+		TypeID:    tsInfo.param.TypeID,
+		ParamID:   tsInfo.param.ParamID,
+		Sensor:    &tsInfo.param.Sensor,
+		Level:     tsInfo.param.Hlevel,
+	}
+	tsid, err := lard.GetTimeseriesID(label, tsInfo.param.Fromtime, pool)
+	if err != nil {
+		slog.Error(tsInfo.logstr + "could not obtain timeseries - " + err.Error())
+		return 0, err
+
+	}
+	return tsid, nil
+}
+
+func (table *Table) parseData(handle *os.File, meta *TimeseriesInfo, config *ImportConfig) ([]LardObs, error) {
+	scanner := bufio.NewScanner(handle)
+
+	var rowCount int
+	// Try to infer row count from header
+	if config.HasHeader {
+		scanner.Scan()
+		// rowCount, _ = strconv.Atoi(scanner.Text())
+		if temp, err := strconv.Atoi(scanner.Text()); err == nil {
+			rowCount = temp
+		}
+	}
+
+	data := make([]LardObs, 0, rowCount)
+	for scanner.Scan() {
+		cols := strings.Split(scanner.Text(), config.Sep)
+
+		obsTime, err := time.Parse("2006-01-02_15:04:05", cols[0])
+		if err != nil {
+			return nil, err
+		}
+
+		// Only import data between KDVH's defined fromtime and totime
+		if meta.span.FromTime != nil && obsTime.Sub(*meta.span.FromTime) < 0 {
+			continue
+		} else if meta.span.ToTime != nil && obsTime.Sub(*meta.span.ToTime) > 0 {
+			break
+		}
+
+		if obsTime.Year() >= table.importUntil {
+			break
+		}
+
+		temp, err := table.convFunc(Obs{meta, obsTime, cols[1], cols[2]})
+		if err != nil {
+			return nil, err
+		}
+
+		data = append(data, temp)
+	}
+
+	return data, nil
+}
+
+// TODO: add CALL_SIGN? It's not in stinfosys?
```

Comment:
T_MDATA (basically the only table I've been testing against). I'll add this info to the code!

Reply:
Now that I think about it, if T_MDATA (or any other table) contains moving stations we probably should not store it in the same table we store static stations, since they require additional metadata (position?). But we haven't decided on the table structure yet.

Reply:
Non-moving installations like oil platforms also have call signs, and they make up most of that table, I think. We do also have ship data, and there are paramid for MLON and MLAT which contains timeseries of the coordinates of these stations. That is how it was solved in KDVH. So if you see MLAT and MLON, that means there could be ship data in a table. (The table T_CDCV_DATA also contains data from buoys, though I don't remember if they move around a lot.)